pkgs/build-support/fetchgithub/default.nix

```
@@ -44,7 +44,10 @@ let
     else fetchzip;
   privateAttrs = lib.optionalAttrs private {
     netrcPhase =
-      let machineName = if githubBase == "github.com" then "api.github.com" else githubBase;
+      # When using private repos:
+      # - Fetching with git works using https://github.com but not with the GitHub API endpoint
+      # - Fetching a tarball from a private repo requires to use the GitHub API endpoint
+      let machineName = if githubBase == "github.com" && !forceFetchGit then "api.github.com" else githubBase;
```

```suggestion
      let machineName = if githubBase == "github.com" && !useFetchGit then "api.github.com" else githubBase;
```

The issue here is going to be that, when we're using `fetchgit`, we're setting credentials for `api.github.com` rather than `github.com`; whether we're using `fetchgit` is determined by the value of `useFetchGit`, and `forceFetchGit` is only one of the reasons we might do that.

---

Indeed you are right, I was so focus on fixing my own problem that I didn't have a broader look!